pymodbus/client/tcp.py

```diff
@@ -86,8 +86,16 @@ def __init__(  # pylint: disable=too-many-arguments
         )
 
     def close(self, reconnect: bool = False) -> None:
-        """Close connection."""
```

Comment:
This is wrong, this method is called by the app when it wants to close the connection.

Why would you prevent the app from closing the connection ? Your problem is down in the transport layer.

Reply:
Not sure how this can be resolved in the transport layer. The issue is the call to close() in async_execute() in client/base.py after too many retries of a particular request. The remote end will never respond to this request as it addresses an inactive slave. It will however respond to other requests, and closing the connection as a result causes these to fail too.

Reply:
if the issue if the call to close in async_execute, then that would be a better place to fix the problem rather than prohibiting the app to close a connection.

Be aware your PR highlighted some old code, that should have been updated long time ago, it will be so very shortly.